earthquakes.py

```diff
@@ -24,38 +27,85 @@ def get_data():
     # The response we get back is an object with several fields.
     # The actual contents we care about are in its text field:
     text = response.text
+    with open("downloaded.json", "w") as output_file:
+        output_file.write(response.text)
+    data = json.loads(response.text)
+    print(type(data))
+    for key in data:
+        print(key)
+    #with open("rewritten.json", "w") as new_output_file:
+    #    json.dump(data, new_output_file)
+
     # To understand the structure of this text, you may want to save it
     # to a file and open it in VS Code or a browser.
     # See the README file for more information.
     ...
 
     # We need to interpret the text to get values that we can work with.
     # What format is the text in? How can we load the values?
-    return ...
+    return data
 
 def count_earthquakes(data):
-    """Get the total number of earthquakes in the response."""
-    return ...
+    return data.get("metadata").get("count")
 
 
 def get_magnitude(earthquake):
     """Retrive the magnitude of an earthquake item."""
-    return ...
+    return [dict.get("properties").get("mag") for dict in data.get("features")]
 
 
 def get_location(earthquake):
     """Retrieve the latitude and longitude of an earthquake item."""
     # There are three coordinates, but we don't care about the third (altitude)
-    return ...
+    list_of_coords = [dict.get("geometry").get("coordinates") for dict in data.get("features")]
+    lat_lon_only = [coords[0:2] for coords in list_of_coords]
+    return lat_lon_only
 
 
 def get_maximum(data):
     """Get the magnitude and location of the strongest earthquake in the data."""
-    ...
-
+    mag_lat_lon = sorted([(([dict.get("properties").get("mag")]+dict.get("geometry").get("coordinates"))[0:3]) for dict in data.get("features")], reverse = True)
+    max_magnitude = mag_lat_lon[0][0]
+    max_location = mag_lat_lon[0][1:3]
+    return max_magnitude, max_location
 
 # With all the above functions defined, we can now call them and get the result
 data = get_data()
+
 print(f"Loaded {count_earthquakes(data)}")
 max_magnitude, max_location = get_maximum(data)
-print(f"The strongest earthquake was at {max_location} with magnitude {max_magnitude}")
\ No newline at end of file
+print(f"The strongest earthquake was at {max_location} with magnitude {max_magnitude}")
+
+
+############################################################################################
+############################################################################################
+
+#the frequency (number) of earthquakes per year
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import datetime
+time_humanfriendly = [datetime.datetime.fromtimestamp((dict.get("properties").get("time"))/1000).strftime('%Y') for dict in data.get("features")]
```

Comment:
`strftime` will convert the date (in this case, the year only) to a string. I think it would be better to keep the year as an integer, which you can also do by calling `.year` on the `datetime` object.